core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -127,6 +136,19 @@ public void setFileIOBuilder(Function<Map<String, String>, FileIO> newIOBuilder)
     this.ioBuilder = newIOBuilder;
   }
 
+  public void setClientPoolBuilder(
```

Comment:
The IO builder seemed like a reasonable thing to get in quickly, but it does conflict with the `initialize` pattern. If we are going to inject more resources then we should come up with a standard way to do that rather that creating a bunch of these setter methods.

I think what we should probably do is inject these hooks in the constructor. Would that work for you, @electrum?

Reply:
Agreed, constructor injection is better. I was trying to follow the existing pattern.

Reply:
Yeah, let's switch over to constructor. I wasn't sure what would work for Trino originally but that's a better pattern.